libs/event-manager/web-client/features/profile/src/lib/user.store.ts

```diff
@@ -0,0 +1,94 @@
+import { Injectable } from '@angular/core';
+import { ComponentStore, tapResponse } from '@ngrx/component-store';
+import { PublicKey } from '@solana/web3.js';
+import { FirebaseService } from 'libs/event-manager/web-client/data-access/src/lib/firebase/firebase.service';
+import { combineLatest, defer, EMPTY, from, map, switchMap } from 'rxjs';
+
+interface UserState {
+  id: PublicKey | null;
+  user: User | null;
+  loading: boolean;
+  error: unknown | null;
+}
+
+export interface User {
+  name: string;
+  lastName: string;
+  email: string | null;
+  image: string | null;
+  discoTokens: number;
+}
+
+const initialState: UserState = {
+  user: null,
+  loading: false,
+  id: null,
+  error: null,
+};
+
+@Injectable()
+export class UserStore extends ComponentStore<UserState> {
+  readonly user$ = this.select(({ user }) => user);
+  readonly id$ = this.select(({ id }) => id);
+  readonly error$ = this.select(({ error }) => error);
+
+  constructor(private readonly _firebaseService: FirebaseService) {
+    super(initialState);
+
+    this._loadUser(
+      combineLatest([
+        // Trigger load events when connection changes
+        this.select(this.id$, (id) => ({ id }), { debounce: true }),
+      ]).pipe(map(([data]) => data))
+    );
+  }
+
+  readonly setUserId = this.updater<PublicKey | null>((state, id) => ({
+    ...state,
+    id: id,
+  }));
+
+  private readonly _loadUser = this.effect<{
+    id: PublicKey | null;
+  }>(
+    switchMap(({ id }) => {
+      // If there's no connection ignore loading call
+      if (id === null) {
+        return EMPTY;
+      }
+
+      this.patchState({ loading: true });
+
+      return this._firebaseService.getUser(id.toBase58()).pipe(
+        tapResponse(
+          (user) => {
+            if (user != undefined)
```

Comment:
always prefer [strict equality rather than abstract equality](https://stackoverflow.com/questions/359494/which-equals-operator-vs-should-be-used-in-javascript-comparisons#:~:text=The%20%3D%3D%20operator%20will%20compare,Both%20are%20equally%20quick.)